### PR TITLE
[TACHYON-1207] Added tests for lineage options and TachyonLineage

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-1207

### CODE REVIEW

File: clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  /**
+   * Tests that building a {@link DeleteLineageOptions} works.
+   */
+  @Test
+  public void builderTest() {
+    DeleteLineageOptions options =
+        new DeleteLineageOptions.Builder(new TachyonConf()).setCascade(true).build();
+    Assert.assertTrue(options.isCascade());
+  }
```
Comment (calvinjia) [inline clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java]: Do you mind adding this after we finish TACHYON-167?

**Comment by yupeng9** [comment]
Jenkins, test this please

File: clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java (hunk 1 above)
Comment (yupeng9) [inline clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java]: okay, no problem

File: clients/unshaded/src/test/java/tachyon/client/lineage/TachyonLineageTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  private LineageContext mLineageContext;
+  private LineageMasterClient mLineageMasterClient;
+  private TachyonLineage mTachyonLineage;
+  private TachyonConf mTachyonConf;
+
+  @Before
+  public void before() throws Exception {
+    mTachyonConf = new TachyonConf();
+    mTachyonConf.set(Constants.USER_LINEAGE_ENABLED, "true");
+    PowerMockito.mockStatic(ClientContext.class);
```
Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/lineage/TachyonLineageTest.java]: We can use `ClientContext.reset(mTachyonConf)` here and `ClientContext.reset()` in an `after()`

Comment (yupeng9) [inline clients/unshaded/src/test/java/tachyon/client/lineage/TachyonLineageTest.java]: done.

File: clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java (hunk 1 above)
Comment (yupeng9) [inline clients/unshaded/src/test/java/tachyon/client/lineage/options/DeleteLineageOptionsTest.java]: When do you plan to merge the client branch?

**Comment by aaudiber** [comment]
LGTM

**Comment by aaudiber** [comment]
LGTM, just need to fix merge conflicts

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**